esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -205,6 +205,56 @@ WsWUExceptions::WsWUExceptions(IConstWorkUnit& wu): numerr(0), numwrn(0), numinf
         errors.append(*e.getLink());
     }
 }
+#ifdef _CONTAINERIZED
+void WsWuInfo::readWorkunitComponentLogs(MemoryBuffer& buf, const char* outFile, unsigned maxLogRecords = 10000,
+                                         LogAccessReturnColsMode retColsMode = RETURNCOLS_MODE_default, LogAccessLogFormat logFormat = LOGACCESS_LOGFORMAT_csv)
+{
+    if (!m_remoteLogAccessor)
+        throw makeStringException(-1, "WsWuInfo: Remote Log Access plug-in not available!");
+
+    LogAccessConditions logFetchOptions;
+    logFetchOptions.setFilter(getJobIDLogAccessFilter(wuid.str()));
+
+    StringBuffer start, end;
+    stat_type timeStamp;
+    if (cw->getStatistic(timeStamp, "", StWhenCreated))
+    {
+        formatStatistic(start.clear(), (timeStamp-6000000), SMeasureTimestampUs); //start - time buffer
+        DBGLOG("WUID '%s' started (adjusted)'%s'", wuid.str(), start.str());
+    }
+
+    if (cw->getStatistic(timeStamp, "", StWhenFinished))
+    {
+        formatStatistic(end.clear(), (timeStamp+6000000), SMeasureTimestampUs).str();
+        DBGLOG("WUID '%s' ended (adjusted) '%s'", wuid.str(), end.str()); //end + time buffer
+    }
+
+    struct LogAccessTimeRange range;
+    range.setStart(start.str());
```

Comment:
setStart requires a const char *

Reply:
The StringBuffer start will be converted automatically to const char * by StringBuffer's inline operator.

Reply:
Thanks @wangkx other than not being ideal for maintenance, is there an advantage worth making the change?

Reply:
It makes the code clear. I got warnings from code reviewers when I left the str() inside my code.

Reply:
@wangkx thanks for clarification. It seems like a personal preference. It might be clear if the developer/maintainer has deep knowledge of The StringBuffer inline operator. The explicit .str() function call makes the code explicit and self documenting, hence without an added advantage in performance, we'll keep the code as is. Please note your code review conclusion